go-selinux/selinux_linux.go

```diff
@@ -410,6 +427,13 @@ func CanonicalizeContext(val string) (string, error) {
 	return readWriteCon(filepath.Join(getSelinuxMountPoint(), "context"), val)
 }
 
+/*
+ComputeCreateContext requests the type transition from source to target for class  from the kernel.
+*/
+func ComputeCreateContext(source string, target string, class int) (string, error) {
```

Comment:
I'd prefer a bit higher-level interface -- pass the class as a string (e.g. "file") and do a conversion to index internally. It's easy to do it with caching and map.